Private data edits (#1960)
hyperledger/fabric: docs/source/private_data_tutorial.rst

    }
 ]

The ``policy`` property in the ``assetCollection`` definition specifies that both
Org1 and Org2 can store the collection on their peers. The ``memberOnlyRead``
and ``memberOnlyWrite`` parameters are used to specify that only Org1 and

> Org2 can read and write to this collection.

Reviewer: People are always confused about whether memberOnlyWrite applies to both client and endorsing peer... let's clarify that we're talking about client calls for these properties...

```suggestion
Org2 clients can read and write to this collection.
```

Author: done.

> The ``Org1MSPPrivateCollection`` collection allows only members of Org1 to have

Reviewer: ```suggestion
The ``Org1MSPPrivateCollection`` collection allows only peers of Org1 to have
```

Author: done. made changes to other places as well.

the private data in their private database, while the ``Org2MSPPrivateCollection``

> collection can only be stored by members of Org2. The ``endorsementPolicy`` parameter

Reviewer: ```suggestion
collection can only be stored by peers of Org2. The ``endorsementPolicy`` parameter
```

Author: done.

Query private data Org2 is not authorized to
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Peers in Org2 do not have the Org1 asset1 ``appraisedValue`` private data in their

Reviewer 2: ```suggestion
Peers in Org2 do not have the Org1 owned asset1's ``appraisedValue`` private data in their
```

Author: Peers in Org2 do not have the ``appraisedValue`` of the asset1 owned by Org2 in their
side database.

> side database. Run the following command to demonstrate that the asset appraisal

Reviewer 2: ```suggestion
side database. Run the following command, to demonstrate that the asset's ``appraisalValue``
```

Author: done.

The empty response shows that the asset1 private details do not exist in buyer
(Org2) private collection.

Nor can a user from Org2 read the Org1 private data collection:

.. code:: bash

    peer chaincode query -C mychannel -n private -c '{"function":"ReadAssetPrivateDetails","Args":["Org1MSPPrivateCollection","asset1"]}'

By setting ``"memberOnlyRead": true`` in the collection configuration file, we
specify that only clients from Org1 can read data from the collection. An Org2 client
who tries to read the collection would only get the following response:

.. code:: json

    Error: endorsement failure during query. response: status:500 message:"failed to
    read asset details: GET_STATE failed: transaction ID: d23e4bc0538c3abfb7a6bd4323fd5f52306e2723be56460fc6da0e5acaee6b23: tx
    creator does not have read access permission on privatedata in chaincodeName:private collectionName: Org1MSPPrivateCollection"

> Only users from Org2 will only be able to see the public hash of the private data.

Reviewer: ```suggestion
Users from Org2 will only be able to see the public hash of the private data.
```

Author: dumb mistake as I was going through last night. Fixed.

Query private data Org2 is not authorized to
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Peers in Org2 do not have the ``appraisedValue`` of the asset1 owned by Org2 in their

Reviewer 2: On reading this sentence again, I think the line is confusing since 

 asset1 is still owned by Org1 (transfer hasn't happened yet) ... the peer is of course owned by Org2.

Rephrase, or perhaps delete the "owned" part

Author: Rephrasing: 
Because the asset was created by Org1, peers in Org2 do not store the
``appraisedValue`` associated with ``asset1``. Run the following command to
demonstrate that the asset's ``appraisedValue`` is not stored in the
``Org2MSPPrivateCollection`` on the Org2 peer:

Reviewer 2: sweet

Query private data Org2 is not authorized to
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Because the asset was created by Org1, peers in Org2 do not store the

Reviewer: ```suggestion
Because the asset was created by Org1 and stored in the ``Org1MSPPrivateCollection`` collection, peers in Org2 do not store the
```

Author: Because the asset was created by Org1, the ``appraisedValue`` associated with
``asset1`` is stored in the ``Org1MSPPrivateCollection`` collection. The value is
not stored by peers in Org2.

``appraisedValue`` associated with ``asset1``. Run the following command to
demonstrate that the asset's ``appraisedValue`` is not stored in the
``Org2MSPPrivateCollection`` on the Org2 peer:

:guilabel:`Try it yourself`

.. code:: bash

    peer chaincode query -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile ${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem -C mychannel -n private -c '{"function":"ReadAssetPrivateDetails","Args":["Org2MSPPrivateCollection","asset1"]}'

The empty response shows that the asset1 private details do not exist in buyer
(Org2) private collection.

> Nor can a member of Org2 read the Org1 private data collection:

Reviewer: ```suggestion
Nor can a client of Org2 read the Org1 private data collection:
```

Author: Nor can a user from Org2 read

.. code:: bash

    peer chaincode query -C mychannel -n private -c '{"function":"ReadAssetPrivateDetails","Args":["Org1MSPPrivateCollection","asset1"]}'

By setting ``"memberOnlyRead": true`` in the collection configuration file, we

> specify that only members of Org1 can read data from the collection. An Org2 member

Reviewer: ```suggestion
specify that only clients of Org1 can read data from the collection. An Org2 client
```

Author: done.

> who tries to read the collection would only get the following response:

Reviewer: Remove the next line.

:guilabel:`Try it yourself`

> Since we need to keep track of how many blocks we add to see if our private data gets purged, open a new terminal window and view the private data logs for this peer by

Reviewer 2: thanks for cleanup !

:guilabel:`Try it yourself`

Since we need to keep track of how many blocks we add to see if our private data gets purged, open a new terminal window and view the private data logs for this peer by
running the following command. Note the highest block number.

.. code:: bash

    docker logs peer0.org1.example.com 2>&1 | grep -i -a -E 'private|pvt|privdata'

Switch back to the terminal with our peer CLI. Now that we are operating
as a member of Org2, we can demonstrate that the asset appraisal is not stored
in Org2MSPPrivateCollection, on the Org2 peer:

.. code:: bash

    peer chaincode query -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile ${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem -C mychannel -n private -c '{"function":"ReadAssetPrivateDetails","Args":["Org2MSPPrivateCollection","asset1"]}'

> The empty response shows that the asset1 private details do not exist in buyer (Org2) private collection.

Author: This queries were repetitive with the section above, so I consolidated the sections using the language in this section and removed the redundant queries.

:guilabel:`Try it yourself`

> Switch back to your other terminal window run the following commands to operate

Reviewer 2: ```suggestion
Switch back to your other terminal window and run the following commands to operate
```

Author: done.
